src/client/results/mostUsedWordsReducer.js

```diff
@@ -1,7 +1,23 @@
-const MostUsedWordsReducer = (state = { 'filterTerm': '' }, action) => {
+import { HIDE_MOST_USED_WORD, SHOW_MOST_USED_WORD } from './mostUsedWordsActions';
+
+const MostUsedWordsReducer = (state = { filterTerm: '', wordsToHide: {} }, action) => {
   switch (action.type) {
     case 'UPDATE_SEARCH_MOST_USED_WORDS_FILTER':
-      return { ...state, 'filterTerm': action.filterTerm };
+      return { ...state, filterTerm: action.filterTerm };
+    case HIDE_MOST_USED_WORD:
+      {
+        const result = { ...state, wordsToHide: Object.assign({}, state.wordsToHide) };
+        result.wordsToHide[action.word] = true;
+        return result;
+      }
+
+    case SHOW_MOST_USED_WORD:
+      {
+        const result = { ...state, wordsToHide: Object.assign({}, state.wordsToHide) };
+        delete result.wordsToHide[action.word];

```

Comment:
Removes the key entirely from the object, as opposed to simply setting it to undefined. Required because later on I iterate through the keys of the object.